### PR TITLE
chore: added v28 guide

### PR DESCRIPTION
**Pull request by crnbarr93**
## What is the purpose of the change

Add v28 binaries json and upgrade guide

### CODE REVIEW

File: networks/osmosis-1/upgrades/v28/mainnet/guide.md [hunk 1]
```diff
@@ -0,0 +1,116 @@
+# Mainnet Upgrade Guide: From Version v26 to v28
```
Comment (niccoloraspa) [inline networks/osmosis-1/upgrades/v28/mainnet/guide.md:1]: v27 to v28

**Review by niccoloraspa — CHANGES_REQUESTED** [review]

File: networks/osmosis-1/upgrades/v28/mainnet/v28_binaries.json [hunk 2]
```diff
@@ -0,0 +1,6 @@
+{
+  "binaries": {
+    "linux/amd64": "https://github.com/osmosis-labs/osmosis/releases/download/v28.0.0/osmosisd-28.0.0-linux-amd64?checksum=TODO",
+    "linux/arm64": "https://github.com/osmosis-labs/osmosis/releases/download/v28.0.0/osmosisd-28.0.0-linux-arm64?checksum=TODO"
```
Comment (niccoloraspa) [inline networks/osmosis-1/upgrades/v28/mainnet/v28_binaries.json:4]: Let's do the release first and get the proper checksum in place

File: networks/osmosis-1/upgrades/v28/mainnet/guide.md [hunk 3]
```diff
@@ -0,0 +1,116 @@
...
+```sh
+go install github.com/cosmos/cosmos-sdk/cosmovisor/cmd/cosmovisor@v1.0.0
+mkdir -p ~/.osmosisd
+mkdir -p ~/.osmosisd/cosmovisor
+mkdir -p ~/.osmosisd/cosmovisor/genesis
+mkdir -p ~/.osmosisd/cosmovisor/genesis/bin
+mkdir -p ~/.osmosisd/cosmovisor/upgrades
+cp $GOPATH/bin/osmosisd ~/.osmosisd/cosmovisor/genesis/bin
+mkdir -p ~/.osmosisd/cosmovisor/upgrades/v26/bin
+cp $GOPATH/bin/osmosisd ~/.osmosisd/cosmovisor/upgrades/v26/bin
```
Comment (niccoloraspa) [inline networks/osmosis-1/upgrades/v28/mainnet/guide.md:61]: path is wrong `/v27/`

**Comment by niccoloraspa** [comment]
There is also this `upgrade_guide` script to generate the upgrade guide: https://github.com/osmosis-labs/osmosis/tree/main/scripts/release/create_upgrade_guide

**Comment by niccoloraspa** [comment]
This is the GUIDE that got generated:

```markdown
# Mainnet Upgrade Guide: From Version v27 to v28

## Overview

- **v28 Proposal**: [Proposal Page](https://www.mintscan.io/osmosis/proposals/T.B.D)
- **v28 Upgrade Block Height**: T.B.D
- **v28 Upgrade Countdown**: [Block Countdown](https://www.mintscan.io/osmosis/blocks/T.B.D)

## Hardware Requirements

### Memory Specifications

Although this upgrade is not expected to be resource-intensive, a minimum of 64GB of RAM is advised. If you cannot meet this requirement, setting up a swap space is recommended.

#### Configuring Swap Space

*Execute these commands to set up a 32GB swap space*:

```sh
sudo swapoff -a
sudo fallocate -l 32G /swapfile
sudo chmod 600 /swapfile
sudo mkswap /swapfile
sudo swapon /swapfile
```

*To ensure the swap space persists after reboot*:

```sh
sudo cp /etc/fstab /etc/fstab.bak
echo '/swapfile none swap sw 0 0' | sudo tee -a /etc/fstab
```

For an in-depth guide on swap configuration, please refer to [this tutorial](https://www.digitalocean.com/community/tutorials/how-to-add-swap-space-on-ubuntu-20-04).

---

## Cosmovisor Configuration

### Initial Setup (For First-Time Users)

If you have not previously configured Cosmovisor, follow this section; otherwise, proceed to the next section.

Cosmovisor is strongly recommended for validators to minimize downtime during upgrades. It automates the binary replacement process according to on-chain `SoftwareUpgrade` proposals.

Documentation for Cosmovisor can be found [here](https://docs.cosmos.network/main/tooling/cosmovisor).

#### Installation Steps

*Run these commands to install and configure Cosmovisor*:

```sh
go install github.com/cosmos/cosmos-sdk/cosmovisor/cmd/cosmovisor@v1.0.0
mkdir -p ~/.osmosisd
mkdir -p ~/.osmosisd/cosmovisor
mkdir -p ~/.osmosisd/cosmovisor/genesis
mkdir -p ~/.osmosisd/cosmovisor/genesis/bin
mkdir -p ~/.osmosisd/cosmovisor/upgrades
cp $GOPATH/bin/osmosisd ~/.osmosisd/cosmovisor/genesis/bin
mkdir -p ~/.osmosisd/cosmovisor/upgrades/v27/bin
cp $GOPATH/bin/osmosisd ~/.osmosisd/cosmovisor/upgrades/v27/bin
```

*Add these lines to your profile to set up environment variables*:

```sh
echo "# Cosmovisor Setup" >> ~/.profile
echo "export DAEMON_NAME=osmosisd" >> ~/.profile
echo "export DAEMON_HOME=$HOME/.osmosisd" >> ~/.profile
echo "export DAEMON_ALLOW_DOWNLOAD_BINARIES=false" >> ~/.profile
echo "export DAEMON_LOG_BUFFER_SIZE=512" >> ~/.profile
echo "export DAEMON_RESTART_AFTER_UPGRADE=true" >> ~/.profile
echo "export UNSAFE_SKIP_BACKUP=true" >> ~/.profile
source ~/.profile
```

### Upgrading to v28

*To prepare for the upgrade, execute these commands*:

```sh
mkdir -p ~/.osmosisd/cosmovisor/upgrades/v28/bin
cd $HOME/osmosis
git pull
git checkout v28.0.0
make build
cp build/osmosisd ~/.osmosisd/cosmovisor/upgrades/v28/bin
```

At the designated block height, Cosmovisor will automatically upgrade to version v28.

---

## Manual Upgrade Procedure

Follow these steps if you opt for a manual upgrade:

1. Monitor Osmosis until it reaches the specified upgrade block height: T.B.D.
2. Observe for a panic message followed by continuous peer logs, then halt the daemon.
3. Perform these steps:

```sh
cd $HOME/osmosis
git pull
git checkout v28.0.0
make install
```

4. Restart the Osmosis daemon and observe the upgrade.

---

## Additional Resources

- Osmosis Documentation: [Website](https://docs.osmosis.zone)
- Community Support: [Discord](https://discord.gg/pAxjcFnAFH)
```

File: networks/osmosis-1/upgrades/v28/mainnet/v28_binaries.json (hunk 2 above)
Comment (PaddyMc) [reply networks/osmosis-1/upgrades/v28/mainnet/v28_binaries.json:4]: :heavy_plus_sign:

**Review by PaddyMc — APPROVED** [review]

**Review by niccoloraspa — APPROVED** [review]

**Pull request merged**